Marlin/src/HAL/platforms.h

```diff
@@ -38,6 +38,7 @@
 #elif defined(__STM32F1__) || defined(TARGET_STM32F1)
   #define HAL_PATH(PATH, NAME) XSTR(PATH/STM32F1/NAME)
 #elif defined(ARDUINO_ARCH_STM32)
```

Comment:
may need !defined(STM32GENERIC) here ;) i see it listed in your Arduino IDE screenshot...

Reply:
I'm guessing either `__STM32F1__` or `TARGET_STM32F1` is defined in the build flags for STM32GENERIC just as it is with Maple, assuming it's appropriate to build it with `HAL/STM32F1`. That said, I'm not really sure we still support STM32GENERIC at this point because it is not being updated, and STM32duino seems to be getting updates pretty frequently these days.

Reply:
right, STM32GENERIC "may" use STM32F1 HAL (for F1 chips, but really unsure its compatible), its named GENERIC because it was a first attempt to support also F4 & F7... maybe some others too.  All fine, need to be ignored imo ;)